### PR TITLE
spring: 105.0.1-1486-g8581792 -> 106.0

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Description of changes

Related: https://github.com/NixOS/nixpkgs/pull/201859. Change `python` to `python3`.

The previous derivation was also not working since it was using `gcc10StdenvCompat` while linking to a GCC 11 compiled library. Compiling this with current `stdenv` works fine though, and also removed the need of some patches (like the removal of `constexpr`).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/spring/default.nix [hunk 1]
```diff
@@ -68,41 +56,36 @@ stdenv.mkDerivation rec {
...
   ]
-  ++ lib.optional withAI jdk
-  ++ lib.optional withAI python;
-
-  NIX_CFLAGS_COMPILE = "-fpermissive"; # GL header minor incompatibility
+  ++ lib.optionals withAI [ python3 jdk ];
 
   postInstall = ''
     wrapProgram "$out/bin/spring" \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ stdenv.cc.cc systemd ]}"
```
Comment (thiagokokada) [inline pkgs/games/spring/default.nix:98]: Searched in the repo and didn't find any mention of systemd usage except for an unrelated service that they use in their buildbot. Also the games looks like it works fine without it.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 205214` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>

File: pkgs/games/spring/default.nix [hunk 2]
```diff
@@ -1,63 +1,51 @@
...
-  # taken from https://github.com/spring/spring/commits/maintenance
-  src = fetchFromGitHub {
-    owner = "spring";
-    repo = pname;
-    inherit rev;
-    sha256 = "05lvd8grqmv7vl8rrx02rhl0qhmm58dyi6s78b64j3fkia4sfj1r";
-    fetchSubmodules = true;
+  src = fetchurl {
+    url = "https://springrts.com/dl/buildbot/default/master/${version}/source/spring_${version}_src.tar.gz";
+    sha256 = "sha256-mSA4ioIv68NMEB72lYnwDb1QOuWr1VHwu4+grAoHlV0=";
```
Comment (thiagokokada) [inline pkgs/games/spring/default.nix:36]: Using the releases avoid the costly `fetchSubmodules` usage and also removes the need of generating the version ourselves.

File: pkgs/games/spring/default.nix [hunk 3]
```diff
@@ -1,63 +1,51 @@
...
   };
 
-  # The cmake included module correcly finds nix's glew, however
-  # it has to be the bundled FindGLEW for headless or dedicated builds
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/games/spring/default.nix:39]:
```suggestion
  postPatch = ''
```

File: pkgs/games/spring/default.nix [hunk 4]
```diff
@@ -1,63 +1,51 @@
...
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/games/spring/default.nix:40]: Which files exactly?

File: pkgs/games/spring/default.nix [hunk 5]
```diff
@@ -68,41 +56,37 @@ stdenv.mkDerivation rec {
...
   '';
 
   meta = with lib; {
     homepage = "https://springrts.com/";
     description = "A powerful real-time strategy (RTS) game engine";
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ qknight domenkozar sorki ];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+    platforms = platforms.unix;
+    broken = with stdenv.hostPlatform; !isx86 || isDarwin;
```
Comment (SuperSandro2000) [inline pkgs/games/spring/default.nix:90]:
```suggestion
    platforms = [ "aarch64-linux" "x86_64-linux" ];
```

Comment (thiagokokada) [reply pkgs/games/spring/default.nix:90]: The game does support other platforms though, and I was hoping that someone with Darwin or ARM expertise would fix this later.

But ok, I will revert back on how it was (BTW, this does not build in `aarch64-linux` yet).

File: pkgs/games/spring/default.nix (hunk 4 above)
Comment (thiagokokada) [reply pkgs/games/spring/default.nix:40]: A lot actually. Almost all shell scripts included have the wrong shebang.

**Comment by thiagokokada** [comment]
@SuperSandro2000 Did the fixes. Also deleted two unused patches that were still in the repo.

**Comment by thiagokokada** [comment]
@ofborg build spring springLobby

File: pkgs/games/spring/default.nix (hunk 5 above)
Comment (thiagokokada) [reply pkgs/games/spring/default.nix]: Actually, upstream removed support for `i686-linux` in the latest release. Will just leave `x86_64-linux` for now.

**Comment by thiagokokada** [comment]
https://github.com/ofborg build spring springLobby

**Comment by thiagokokada** [comment]
@ofborg build spring springLobby

**Comment by thiagokokada** [comment]
@ofborg build spring springLobby

File: pkgs/games/spring/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/games/spring/default.nix:40]: And they are all over the place and not just in one or two directories?

Comment (thiagokokada) [reply pkgs/games/spring/default.nix:40]: Yeah.

**Comment by thiagokokada** [comment]
CC @SuperSandro2000.

**Comment by PedroHLC** [comment]
Result of `nixpkgs-review pr 205214` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>

**Review by PedroHLC — APPROVED** [review]
Confirming it solves the following issue:
```console
error: Function called without required argument "python" at ./nixpkgs/pkgs/games/spring/default.nix:24, did you mean "jython", "python2" or "python3"?
```

**Pull request merged**